src/context/cart-context.jsx

```diff
@@ -1,34 +1,110 @@
+import axios from "axios";
 import { createContext, useContext, useReducer } from "react";
 import { cartReducer } from "../reducer/CartReducer";
 
 const CartContext = createContext();
 
 const CartProvider = ({ children }) => {
+  const encodedToken = localStorage.getItem("token");
 
-  //const encodedToken = localStorage.getItem("token");
-  // useEffect(() => {
-  //   (async () => {
-  //     try {
-  //       const result = await axios.post("/api/user/cart",  {headers: {
-  //         authorization: encodedToken,
-  //       },})
-  //       console.log(result)
-  //       if(result.status === 200){
-  //         dispatch({type:"ADD_TO_CART"})
-  //       }
-  //     } catch (err) {
-  //       console.log(err);
-  //     }
-  //   })();
-  // }, [encodedToken]);
+  const addToCartHandler = async (item) => {
+    try {
+      const response = await axios.post(
+        "/api/user/cart",
+        {
+          product: item,
+        },
+        {
+          headers: {
+            authorization: encodedToken,
+          },
```

Comment:
suggestion : Since you are using this in all your context functions you can use something like const config = { headers: {authorization: encodedToken}} and use it where ever required

Reply:
i will do it